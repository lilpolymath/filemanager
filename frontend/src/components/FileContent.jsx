import React from 'react';
import PropTypes from 'prop-types';

import styles from "./FileContent.module.css";

function FileContent({ file = null }) {
  return (
    <section className={styles.FileContent}>
      <pre>
        {file?.content && atob(file.content)}
      </pre>
    </section>
  );
}

FileContent.propTypes = {
  file: PropTypes.object
};

export default FileContent;
