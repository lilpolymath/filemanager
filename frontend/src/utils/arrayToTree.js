export const arrayToTree = (array) => {
  var map = {}, node, roots = [], i;

  for (i = 0; i < array.length; i += 1) {
    map[array[i].id] = i;
    array[i].children = [];
  }

  for (i = 0; i < array.length; i += 1) {
    node = array[i];

    if (node.parentId) {
      array[map[node.parentId]]?.children.push(node);
    } else {
      roots.push(node);
    }
}

  return roots;
}

export default arrayToTree