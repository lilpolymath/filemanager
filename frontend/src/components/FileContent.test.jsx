import React from 'react';
import { render, screen } from '../utils/test-utils';

import FileContent from './FileContent';

describe('FileContent Component', () => {
  it('should render defaults', () => {
    const { container } = render(
      <FileContent />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
