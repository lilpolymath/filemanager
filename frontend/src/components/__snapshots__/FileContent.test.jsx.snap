// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileContent Component should render defaults 1`] = `
<section
  class="FileContent"
>
  <pre />
</section>
`;
